### PR TITLE
PROF-10073: Read and propagate helm config for profiling

### PR DESCRIPTION
**Pull request by szegedi**
### What does this PR do?

Allows Continuous Profiler to be enabled (or disabled) by Helm Charts config on the controller pod. The cluster-agent, running on the controller pod, will read an environment variable and use this to mutate the configuration of other pods to set the environment variables that will activate profiling within the tracer/profiler client libraries.

This PR follows the same approach as #23618 did for activation of ASM products.

### Motivation

to make it easier for k8s clients to activate Continous Profiling. Simplified installation is a common request.

### Additional Notes

The PR is designed to establish the fundamentals that will make these other PRs work:
* https://github.com/DataDog/helm-charts/pull/1443
* https://github.com/DataDog/datadog-operator/pull/1271
It is necessary to first have the functionality here in the agent before we can make the Helm and Operator changes available. 

Continuous Profiler will have the env var `DD_ADMISSION_CONTROLLER_AUTO_INSTRUMENTATION_APPSEC_ENABLED` set by changes in those Datadog Operator and Helm Charts PRs. It will result in `DD_PROFILING_ENABLED` being propagated to all pods (or those conforming to the filters).

### Possible Drawbacks / Trade-offs

* More complexity in our config handling, to make it easier for customers.

### Describe how to test/QA your changes

* Unit tests have been added and ensured to pass with `invoke test --targets=./pkg/clusteragent`

### CODE REVIEW

File: releasenotes-dca/notes/admission-controller-profiling-fd7678fa28f7b90e.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+features:
+  - |
+    Cluster Agent now supports activating Continuous Profiling
+    via Admission Controller.
```
Comment (maycmlee) [inline releasenotes-dca/notes/admission-controller-profiling-fd7678fa28f7b90e.yaml:12]:
```suggestion
    The Cluster Agent now supports activating Continuous Profiling
    using Admission Controller.
```

**Review by maycmlee — APPROVED** [review]
Just a couple of small suggestions, but approving.

File: releasenotes-dca/notes/admission-controller-profiling-fd7678fa28f7b90e.yaml (hunk 1 above)
Comment (szegedi) [reply releasenotes-dca/notes/admission-controller-profiling-fd7678fa28f7b90e.yaml]: Thank you for the suggestion, updated!

**Review by eliottness — APPROVED** [review]
LGTM

**Review by ogaca-dd — APPROVED** [review]
LGTM for files owned by ASC

**Review by adel121 — APPROVED** [review]
Approved for #container-platform owned files. 

Some suggestions:
- It would be nice to have some E2E tests to validate that injecting this env var doesn't cause conflicts or issues in the nominal case of a user app. We already have e2e tests for auto instrumentation and language detection [here](https://github.com/DataDog/datadog-agent/blob/main/test/new-e2e/tests/language-detection/language_detection_test.go), so you can augment them to test your change too. 
- It would be good to remove the label `team/container-platform` from this PR because the QA should not be done by our team, but rather by the team that is directly impacted by this feature. Our team scope is only ensuring the admission controller injects what is expected, but QA here should ensure that feature works fine.

**Comment by szegedi** [comment]
Thanks for the suggestion @adel121! There's currently an e2e onboarding system tests [draft PR](https://github.com/DataDog/system-tests/pull/2691) that exercises this functionality (with the drawback that it can only run tests for a released agent.) I'm glad to see that there's an AWS/Pulumi framework locally in the agent as well. I learned since from @robertomonteromiguel that folks that created agent's e2e have helped him create the onboarding system tests infrastructure too. I'll figure out how to validate the feature in this e2e framework and follow up with a test.

**Pull request merged**